File: ts/project/.sdk/tm/zig/core/response.zig
// Transport response wrapper (mirrors go core/response.go). The `json` entry
// stays a function thunk so bodies can be re-read repeatedly.

const h = @import("helpers.zig");
const err = @import("error.zig");
const Value = h.Value;

pub const Response = struct {
    status: i64 = -1,
    status_text: []const u8 = "",
    headers: Value = .{ .null = {} },
    json: Value = .{ .null = {} },
    body: Value = .{ .null = {} },
    err: ?*err.ProjectNameError = null,

    pub fn make(resmap: Value) *Response {
        const r = h.A().create(Response) catch unreachable;
        const status: i64 = switch (h.getp(resmap, "status")) {
            .null => -1,
            else => |s| h.to_int(s),
        };
        r.* = .{
            .status = status,
            .status_text = h.get_str(resmap, "statusText") orelse "",
            .headers = h.getp(resmap, "headers"),
            .json = h.getp(resmap, "json"),
            .body = h.getp(resmap, "body"),
            .err = null,
        };
        return r;
    }
};

File: ts/project/.sdk/tm/rust/core/response.rs
// Transport response wrapper (mirrors go core/response.go). The `json`
// entry stays a Value::Func thunk so bodies can be re-read repeatedly.

use crate::core::error::ProjectNameError;
use crate::core::helpers::{get_str, getp, to_int};
use crate::utility::voxgigstruct::Value;

pub struct Response {
    pub status: i64,
    pub status_text: String,
    pub headers: Value,
    pub json: Value,
    pub body: Value,
    pub err: Option<ProjectNameError>,
}

impl Response {
    pub fn new(resmap: &Value) -> Response {
        let status = match getp(resmap, "status") {
            Value::Noval => -1,
            s => to_int(&s),
        };

        let status_text = get_str(resmap, "statusText").unwrap_or_default();
        let headers = getp(resmap, "headers");
        let json = getp(resmap, "json");
        let body = getp(resmap, "body");

        Response {
            status,
            status_text,
            headers,
            json,
            body,
            err: None,
        }
    }
}
